sdk/storage/azure-storage-blobs/src/blob_container_client.cpp

```diff
@@ -229,16 +229,19 @@ namespace Azure { namespace Storage { namespace Blobs {
         *m_pipeline, m_blobContainerUrl, protocolLayerOptions, context);
   }
 
-  Azure::Response<Models::ListBlobsSinglePageResult> BlobContainerClient::ListBlobsSinglePage(
-      const ListBlobsSinglePageOptions& options,
+  ListBlobsPagedResponse BlobContainerClient::ListBlobs(
+      const ListBlobsOptions& options,
       const Azure::Core::Context& context) const
   {
-    _detail::BlobRestClient::BlobContainer::ListBlobsSinglePageOptions protocolLayerOptions;
+    _detail::BlobRestClient::BlobContainer::ListBlobsOptions protocolLayerOptions;
     protocolLayerOptions.Prefix = options.Prefix;
-    protocolLayerOptions.ContinuationToken = options.ContinuationToken;
+    if (options.ContinuationToken.HasValue() && !options.ContinuationToken.Value().empty())
```

Comment:
> Why don't we set the continuation token for the protocol layer even if the passed in options value is empty?

Yeah, we can do that. That's actually very simple for blob service, because I'm the maintainer of blob auto-generator. But it's not the case for DataLake and File services.